src/coresyftools/coresyftools/tests/test_gpt_tool.py

```diff
@@ -151,3 +152,151 @@ def __call__(self, args):
+
+        with open('input.txt', 'w') as infile:
+            infile.write('input')
+
+        cmd = '--input input.txt --output output --param val'.split()
+        tool.execute(cmd)
+
+        gpt_cmd = 'gpt Land-Sea-Mask -f GeoTIFF-BigTIFF -t {} -param=val {}'.format(
+            os.path.join(os.getcwd(), 'output'), os.path.join(os.getcwd(), 'input.txt.tif')).split()
+        self.assertEqual(call_shell_command_mock.args, gpt_cmd)
+        self.assertTrue(os.path.exists('input.txt.tif'))
+        os.remove('input.txt.tif')
+
+    def test_do_not_add_extension_to_geotiff_when_has_geotiff_extension(self):
+        manifest = self.manifest.copy()
+        manifest['operation'] = {'operation': 'Land-Sea-Mask'}
+        self.write_manifest(manifest)
+        tool = GPTCoReSyFTool(self.runfile)
+
+        class CallShellCommandMock():
+            def __call__(self, args):
+                self.args = args
+
+        call_shell_command_mock = CallShellCommandMock()
+
+        tool._call_shell_command = call_shell_command_mock
+
+        tool._remove_tif_file_extension = lambda x: None
+
+        with open('input.tif', 'w') as infile:
+            infile.write('input')
+
+        cmd = '--input input.tif --output output --param val'.split()
+        tool.execute(cmd)
+
+        gpt_cmd = 'gpt Land-Sea-Mask -f GeoTIFF-BigTIFF -t {} -param=val {}'.format(
+            os.path.join(os.getcwd(), 'output'), os.path.join(os.getcwd(), 'input.tif')).split()
+        self.assertEqual(call_shell_command_mock.args, gpt_cmd)
+        self.assertTrue(os.path.exists('input.tif'))
+        os.remove('input.tif')
+
+
+    def test_do_not_add_extension_to_non_geotiff(self):
+        manifest = self.manifest.copy()
+        manifest['operation'] = {'operation': 'Land-Sea-Mask'}
+        manifest['inputs'] = [{
+                    'identifier': 'input',
+                    'name': 'input',
+                    'description': 'description',
+                    'type': 'Raster'
+                }]
+        self.write_manifest(manifest)
+        tool = GPTCoReSyFTool(self.runfile)
+
+        class CallShellCommandMock():
+            def __call__(self, args):
+                self.args = args
+
+        call_shell_command_mock = CallShellCommandMock()
+
+        tool._call_shell_command = call_shell_command_mock
+
+        tool._remove_tif_file_extension = lambda x: None
+
+        with open('input', 'w') as infile:
+            infile.write('input')
+
+        cmd = '--input input --output output --param val'.split()
+        tool.execute(cmd)
+
+        gpt_cmd = 'gpt Land-Sea-Mask -f GeoTIFF-BigTIFF -t {} -param=val {}'.format(
+            os.path.join(os.getcwd(), 'output'), os.path.join(os.getcwd(), 'input')).split()
+        self.assertEqual(call_shell_command_mock.args, gpt_cmd)
+        self.assertTrue(os.path.exists('input'))
+        os.remove('input')
+
+
+    def test_can_remove_added_extension(self):
```

Comment:
This test does not seem to be correct. I am assuming that the objective is to test the removal of the target/output file extension, in case GPT automatically adds it during execution. If so, the test shall perform the following steps:
1. Add to ```CallShellCommandMock``` the creation of a output file with '.tif' extension, retrieving the output name from the ```args```.
2. Assert if the "output" file exists without extension.

Reply:
You would be correct... if the purpose of the test was what you said. But it is not. :stuck_out_tongue:
The purpose it to ensure that the input gets the original name in the end (the remaining test's are not testing that because we mock the `_remove_tif_extension` ).